src/icons/spritemap.tsx

```diff
@@ -0,0 +1,795 @@
+import React from 'react';
+
+export type IconName = keyof typeof icons;
+
+const icons = {
+  'account-circle': {
```

Comment:
> Not tree-shakeable - (the spritemap.svg also has this issue) all icons end up Webpack bundles regardless of how many icons are actually used

Is this file tree-shakable if we export icons directly one-by-one. For example `export const accountCircle`, `export const addAlarm` instead of exporting one giant default export?

Someone with more insight on icons would have to verify if this is even possible. Im not sure if having the name be `account-circle` instead of `accountCircle` is important.

Reply:
Actually looking more closely it seems like having one big object is important because consumers of the `<Icon />` will pass `name="account-circle"` which is how we know what icon to use.

I'll leave the comment up though

Reply:
> Is this file tree-shakable if we export icons directly one-by-one

Yes, but only if we change the `<Icon>` API (as you noted).

This PR is step 1 (which is no worse than the current state for the most part), and we should talk about whether we want follow on steps to support better tree shaking.

Reply:
2 crumbs for thought:
1. one of the (theoretical) benefits of the separate SVG files was designers could easily add/edit/view them from the codebase. they'll need to do this much less now, but we should still think about how to facilitate that in the new structure
2. do we need tree-shaking? I think the existing spritemap is 60kB large. Brad Frost made this post when we were deciding architecture (march 2022) and there were some suggestions for keeping the `<Icon>` dev-ex while also allowing for tree shaking: https://twitter.com/brad_frost/status/1509160842062929923

Reply:
Interesting twitter poll 🤔 

The developer experience of `<Icon name={...} />` is nice, especially with autocomplete.

Also 60kb (66 for the spritemap.tsx) isn't too bad. Maybe even insignificant in the grand scheme of things.

Reply:
Yah i also think the DX of a `name` prop is very nice. Having done it both ways relatively recently, having separate components is especially useful when there are LOTS of possible complex icons and Illustrations, and it's known that a small subset of them might be used in any given application. That said, we also had a component like `<Icon />` that imported EVERY INDIVIDUAL ICON so ...

Reply:
Maybe the next level would be webpack + esbuild plugins that analyze which `name`s are actually used and prunes the rest 🤯